src/character.cpp

```diff
@@ -6091,17 +6091,17 @@ float Character::rest_quality() const
     return rest;
 }
 
-std::string Character::extended_description() const
+std::vector<std::string> Character::extended_description() const
```

Comment:
Instead of building a vector of strings, you ought to just use `ImGui::Text` and related primitives directly. Each `extended_description` method could simply assume that it is being called inside of a scrollable ImGui child window or other appropriate context.

As currently written you’re allocating a new vector of strings every single frame, which is not great for performance or memory fragmentation.

Reply:
In this case I think that's going fairly deep into premature optimization territory. If there are actual performance concerns, it's pretty easy to cache the vectors since they're not changing anyways. And this way the imgui code is as simple as can be. That means both simpler changes to existing code for this PR and no requirement to learn imgui for future people trying to improve the descriptions.

Reply:
At least cache the result then.